### ISSUE TITLE
Random failures on CircleCI

### ISSUE THREAD

**Issue by metdos** [citusdata/citus#2645]
Time to time we get random failures on CircleCI, we should look into reasons for them. Some examples @velioglu reported:

https://circleci.com/gh/citusdata/citus/2192
https://circleci.com/gh/citusdata/citus/2185
https://circleci.com/gh/citusdata/citus/2175
https://circleci.com/gh/citusdata/citus/2176

**Comment by metdos** [comment]
@velioglu noted that we are not able to merge open master to enterprise master because of similar errors. @jasonmp85 do you have an idea about the reason and the solution to the problem? Is this related to CircleCI or do we have some bugs?

**Comment by jasonmp85** [comment]
Hey @metdos… it's been a while!

I've been hard at work improving the ergonomics of our CircleCI run. In particular, we'll soon be generating test reports for all builds, which will allow us better insight into exactly what has failed, how often, and what tests are slowest.

In addition, I spent _way_ too much time (oops) learning `gdb`—which I didn't have a ton of experience with—to also produce detailed backtraces and source excerpts for any core dumps. I'll be merging this to master today and further CircleCI failures should go much smoother.

**Comment by jasonmp85** [comment]
The follower test failures seem related to us running those tests in the same environment as previous ones. I broke the tests out to their own job last week and have been trying to collect any core dumps ever since but haven't seen a single one.

The `multi_partitioning` failure is simply the matter of ordering the output correctly. @onderkalaci made a change leading up to his unified executor work in 92e87738dd7faeef6e3cebbfbe27df58d62e625f that _might_ have solved it, but I'm still looking out for further failures.

The `could not establish asynchronous connection after 5000 ms` failure was transient; in the future we'll have better metrics around how often that sort of thing occurs. I've bumped the connection timeout in my own branch to address the issue, but I haven't seen any further appearances yet. Closing this issue and will send out an email about my improvements by EoD.
